parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -482,31 +482,39 @@ loop = do
           ppe <- PPE.suffixifiedPPE <$> prettyPrintEnvDecl (Names names0 mempty)
           case (srcle, srclt, mdValuel) of
             (srcle, srclt, [r@(Referent.Ref mdValue)]) -> do
-              mdType <- eval $ LoadTypeOfTerm mdValue
-              case mdType of
-                Nothing -> respond $ MetadataMissingType ppe r
-                Just ty -> do
-                  let steps =
-                        second (const . step $ Type.toReference ty)
-                        . first (Path.unabsolute .  resolveToAbsolute) <$> srcs
-                  let get = Branch.head <$> use root
-                  before <- get
-                  stepManyAt steps
-                  after <- get
-                  (ppe, outputDiff) <- diffHelper before after
-                  respondNumbered $ ShowDiffNamespace
-                     Path.absoluteEmpty Path.absoluteEmpty ppe outputDiff
-                where
-                step mdType b0 = let
-                  tmUpdates terms = foldl' go terms srcle
-                    where
-                    go terms src = op (src, mdType, mdValue) terms
-                  tyUpdates types = foldl' go types srclt
-                    where
-                    go types src = op (src, mdType, mdValue) types
-                  in over Branch.terms tmUpdates . over Branch.types tyUpdates $ b0
-            (_srcle, _srclt, mdValues) ->
-              respond $ MetadataAmbiguous ppe mdValues
+             let defnsAndTerms = Map.fromList $ srcs `zip` (toList . getHQ'Terms <$> srcs)
+             let noReferents = Map.keys $ Map.filterWithKey (\_ v -> null $ v) defnsAndTerms
+             let paths = (\(path', segment) -> Path.snoc' path' $ HQ'.toName segment) <$> noReferents
+             when (not . null $ noReferents) $ respond $ LinkDefnsNotFound paths
+             unless (not. null $ noReferents) $ do
```

Comment:
I realise this probably does not really keep with the rest of the coding style in this file.
Aside from that: 
1. look into the definitions provided by the user (`[(Path', NameSegment.HQSegment)]`). 
2. zip the definitions and the referents for each definition (`Set Referent`). 
3. filter any definitions that do not have any referents: these do not exist in the codebase.